spec/index.bs

```diff
@@ -8640,26 +8640,19 @@ interface GPUCanvasContext {
 {{GPUCanvasContext}} has the following internal slots:
 
 <dl dfn-type=attribute dfn-for="GPUCanvasContext">
-    : <dfn>\[[validConfiguration]]</dfn> of type boolean, initially `false`.
```

Comment:
The `validConfiguration` flag was tracked separately from the `configuration` itself because at the time we did the refactor to fold the swap chain into the context I recall us making the decision that as long as `configure()` had been called (and a device had been supplied) you would always get a texture out of `getCurrentTexture()`, even if it was an invalid one. That normalizes this with how `createTexture()` works, where something like an invalid size would give back an invalid texture in both cases.

Reply:
I see, I had forgotten this. Given that `getCurrentTexture()` can still throw exceptions (if the context entirely unconfigured), I'm not sure how helpful this is?

It would especially make sense for `getCurrentTexture()` to throw if `configure()` also threw, see [1]. Seems to me to be consistent we should either:

1. Have `configure()` _unconditionally_ set the configuration even if it doesn't make sense, and validate asynchronously. If configured incorrectly, return invalid textures from `getCurrentTexture()`.
2. Have `configure()` validate synchronously and on failure either:
     a. Leave the old configuration in place (if any) so `getCurrentTexture()` keeps working as it did.
     b. Leave the canvas unconfigured so `getCurrentTexture()` now throws.

Right now the spec is in an "impossible" in-between state where I think it requires a content-gpu-content round-trip unless the configuration state could somehow move to the gpu process (I don't think it can).

Reply:
> Have configure() unconditionally set the configuration even if it doesn't make sense, and validate asynchronously. If configured incorrectly, return invalid textures from getCurrentTexture().

I suppose since `device` is a required member of `GPUCanvasConfiguration` this makes sense. The IDL validation will throw if they don't provide a `device`, so it satisfies the logic of "if `configure()` threw `getCurrentTexture()` will throw. For everything else we'll be guaranteed to have a device, even if it's a destroyed/invalid one, so we can raise validation errors and create invalid textures.